associations.rb

```diff
@@ -46,26 +47,29 @@ def generate_migrations
 
 
 class Hotel < ActiveRecord::Base
-  #insert our associations here
- 
+  has_many :rooms
+  has_many :bookings, through: :rooms
+  has_many :booked_guests, source: :guest, through: :bookings
+
   def to_s
     "#{name} with #{rooms.count} rooms"
   end
 end
 
 class Booking < ActiveRecord::Base
-  #insert our associations here
-
+  belongs_to :guest, class_name: "User", foreign_key: 'user_id'
+  belongs_to :room, foreign_key: 'room_id'
+  belongs_to :hotel, foreign_key: 'hotel_id'
```

Comment:
When the foreign keys follow this pattern, it will JustWork™️ .  The foreign_key option should only be necessary if your naming pattern breaks convention.